src/atlannot/evaluation.py

```diff
@@ -0,0 +1,210 @@
+
+    Parameters
+    ----------
+    volume: np.ndarray
+        Input volume.
+    region_ids: list[int] | None
+        List of region IDs to compute the jaggedness. If None, the jaggedness
+        is computed for all the region IDs present in the volume.
+    axis: int
+        Axis along which to compute the jaggedness.
+
+    Returns
+    -------
+    results: dict[int, float]
+        Dictionary containing the region id as keys and the mean of the
+        jaggedness of that given region id as values.
+    """
+    metrics = core.compute(volume, coronal_axis_index=axis, regions=region_ids)
+    if region_ids is None:
+        region_ids = sorted(metrics["perRegion"].keys())
+
+    results = {}
+    for region_id in region_ids:
+        results[region_id] = metrics["perRegion"][region_id]["mean"]
+    return results
+
+
+def compute_iou(vol_true, vol_pred, region_ids=None):
+    """Compute the intersection over union of given region IDs.
+
+    Parameters
+    ----------
+    vol_true: np.ndarray
+        Input volume considered as ground truth.
+    vol_pred: np.ndarray
+        Input volume considered as prediction.
+    region_ids: list[int] | None
+        List of region IDs to compute the intersection over union.
+        If None, the IoU is computed for all the region IDs present in the volume.
+
+    Returns
+    -------
+    results: dict[int, float]
+        Dictionary containing the region id as keys and the average of the
+        intersection over union of that given region id as values.
+    """
+    results = {}
+    if region_ids is None:
+        region_ids = np.unique(vol_true)
+
+    for region_id in region_ids:
+        results[region_id] = iou_score(vol_true, vol_pred, k=region_id)[0]
+
+    return results
+
+
+def dist_entropy(data, bins=100):
+    """Compute the entropy of the value distribution of data.
+
+    Parameters
+    ----------
+    data: np.ndarray
+        Data for which to compute the entropy.
+    bins: int
+        Number of bins used to compute the histogram of the densities.
+
+    Returns
+    -------
+    float
+        Entropy of the densities of Nissl at a given region value.
+    """
+    hist, _ = np.histogram(data, bins=bins)
+    return stats.entropy(hist)
+
+
+def compute_conditional_entropy(nissl, atlas):
+    """Compute entropies of Nissl densities.
+
+    Parameters
+    ----------
+    nissl: np.ndarray
+        Nissl volume.
+    atlas: np.ndarray
+        Annotation atlas.
+
+    Returns
+    -------
+    conditional_entropy: float
+        Conditional entropy of the densities of Nissl depending on the brain regions.
+    """
+    n_pixels = (atlas != 0).sum()
+    label_values, count_values = np.unique(atlas, return_counts=True)
+    all_region_entropy = []
+    for label, count in zip(label_values, count_values):
+        all_region_entropy.append(dist_entropy(nissl[atlas == label]) * count)
+
+    conditional_entropy = np.sum(all_region_entropy) / n_pixels
```

Comment:
Just noticed - `np.unique` will also include the background region `label=0`, is that what we want? Especially because at the end we normalize by `n_pixels`, which doesn't include the background.

Reply:
Thanks a lot for catching this! Correction in 936a572.